test/integration-tests/tests/jd_integration.rs

```diff
@@ -108,3 +114,127 @@ async fn jdc_does_not_stackoverflow_when_no_token() {
     tokio::time::sleep(Duration::from_secs(3)).await;
     assert!(tokio::net::TcpListener::bind(jdc_addr).await.is_err());
 }
+
+// The issue arises from the following message flow. JDC first receives a `NewTemplate`,
```

Comment:
This text is not following any convention or any other test doc style. Please look into other tests and how they are documented and amend accordingly.